python/lsst/pipe/base/quantum_graph/_multiblock.py

```diff
@@ -610,6 +610,67 @@ def open_in_zip(cls, zf: zipfile.ZipFile, name: str, int_size: int) -> Iterator[
         with zf.open(f"{name}.mb", mode="r") as stream:
             yield MultiblockReader(stream, int_size)
 
+    @classmethod
+    def read_all_bytes_in_zip(cls, zf: zipfile.ZipFile, name: str, *, int_size: int) -> Iterator[bytes]:
+        """Iterate over all of the byte blocks in a file in a zip archive.
+
+        Parameters
+        ----------
+        zf : `zipfile.ZipFile`
+            Zip archive to read the file from.
+        name : `str`
+            Base name for the multi-block file; an extension will be added.
+        int_size : `int`
+            Number of bytes to use for all integers.
+
+        Returns
+        -------
+        byte_iter : `~collections.abc.Iterator` [ `bytes` ]
+            Iterator over blocks.
+        """
+        with zf.open(f"{name}.mb", mode="r") as zf_stream:
+            # The standard library typing of IO[bytes] tiers isn't consistent.
+            buffered_stream = BufferedReader(zf_stream)  # type: ignore[type-var]
```

Comment:
For buffer sizes set it to a default value that is friendly to the file sitting on the other end of a network transmission, and consider having it set by an environment variable that can be tuned.